modin/numpy/arr.py

```diff
@@ -0,0 +1,333 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+import numpy
+
+from pandas.core.dtypes.common import is_list_like
+from modin.error_message import ErrorMessage
+
+
+class array(object):
+    def __init__(
+        self,
+        object=None,
+        dtype=None,
+        *,
+        copy=True,
+        order="K",
+        subok=False,
+        ndmin=0,
+        like=None,
+        _query_compiler=None,
+        _ndim=None,
+    ):
+        if _query_compiler is not None:
+            self._query_compiler = _query_compiler
+            self._ndim = _ndim
+        elif is_list_like(object) and not is_list_like(object[0]):
+            import modin.pandas as pd
+
+            qc = pd.Series(object)._query_compiler
+            self._query_compiler = qc
+            self._ndim = 1
+        else:
+            expected_kwargs = {
+                "dtype": None,
+                "copy": True,
+                "order": "K",
+                "subok": False,
+                "ndmin": 0,
+                "like": None,
+            }
+            rcvd_kwargs = {
+                "dtype": dtype,
+                "copy": copy,
+                "order": order,
+                "subok": subok,
+                "ndmin": ndmin,
+                "like": like,
+            }
+            for key, value in rcvd_kwargs.copy().items():
+                if value == expected_kwargs[key]:
+                    rcvd_kwargs.pop(key)
+            arr = numpy.array(object, **rcvd_kwargs)
```

Comment:
If we specify the kwargs with default types, we get an error:
```python
In [2]: import numpy as np

In [3]: np.array([1, 2], like=None)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [3], in <cell line: 1>()
----> 1 np.array([1, 2], like=None)

TypeError: The `like` argument must be an array-like that implements the `__array_function__` protocol.
```

Reply:
I did some digging into the numpy source, and it seems that the value numpy uses for optional arguments is `numpy._NoValue` rather than `None` (as [here](https://github.com/numpy/numpy/blob/8cec82012694571156e8d7696307c848a7603b4e/numpy/core/fromnumeric.py#L1250))

Reply:
Replacing `None` with numpy._NoValue and passing it in still causes the code to error out unfortunately.

Reply:
I would still better try to figure proper defaults... since they should be what we should take as function inputs at the frontend layer.

Reply:
Even with the proper defaults that numpy uses, we still error out on passing them through - I think filtering the arguments and only passing the ones that aren't default makes sense to me, although I'm open to suggestions!